### PR TITLE
[5.1] Use update channel the correct way

### PR DESCRIPTION
**Pull request by bembelimen**
Alternatives for #43184

### CODE REVIEW

File: libraries/src/TUF/DatabaseStorage.php [hunk 1]
```diff
@@ -20,8 +20,11 @@
 
 /**
  * @since  5.1.0
+ *
+ * @internal Currently this class is only used for Joomla! updates and will be extended in the future to support 3rd party updates
+ *           Don't extend this class in your own code, it is subject to change without notice.
```
Comment (brianteeman) [inline libraries/src/TUF/DatabaseStorage.php:25]: I thought declaring the class as final meant you cannot extend it. Therefore the sentence is redundant

File: libraries/src/Updater/ConstraintChecker.php [hunk 2]
```diff
@@ -34,16 +38,30 @@ class ConstraintChecker
      */
     protected \stdClass $failedEnvironmentConstraints;
 
+    /**
+     * The channel to check the contrains against
```
Comment (brianteeman) [inline libraries/src/Updater/ConstraintChecker.php]:
```suggestion
     * The channel to check the contraints against
```

Comment (bembelimen) [reply libraries/src/Updater/ConstraintChecker.php]: Thx

File: libraries/src/TUF/DatabaseStorage.php (hunk 1 above)
Comment (bembelimen) [reply libraries/src/TUF/DatabaseStorage.php:25]: Yup, you're right, was a two step thing and I missed the sentence.

**Comment by LadySolveig** [comment]
I have tested this item :white_check_mark: successfully on e9e03f8902eb40eae732634ed03759fcd2fcfec2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/43190">issues.joomla.org/tracker/joomla-cms/43190</a>.</sub>

**Comment by richard67** [comment]
@bembelimen It seems this PR breaks the unit tests for the constraint checker.

File: tests/Unit/Libraries/Cms/Updater/ConstraintCheckerTest.php [hunk 3]
```diff
@@ -14,6 +14,7 @@
 use Joomla\CMS\Updater\ConstraintChecker;
 use Joomla\Database\DatabaseDriver;
 use Joomla\Tests\Unit\UnitTestCase;
+use Joomla\CMS\Version;
```
Comment (Quy) [inline tests/Unit/Libraries/Cms/Updater/ConstraintCheckerTest.php]:
```suggestion
use Joomla\CMS\Updater\ConstraintChecker;
use Joomla\CMS\Version;
use Joomla\Database\DatabaseDriver;
use Joomla\Tests\Unit\UnitTestCase;
```

**Comment by laoneo** [comment]
Why did you revert the final statement? Sounds logic to me instead of the long description that this class is internal only.

**Comment by bembelimen** [comment]
I'm not keen to rewrite and fix all the tests, when I declare them as final (plus, I don't like to close classes that way). Feel free to look into it, but I don't plan to do it before the RC.

**Comment by LadySolveig** [comment]
Thank you!

**Pull request merged**